osbrain/tests/test_agent.py

```diff
@@ -229,6 +260,22 @@ def test_pubsub(nsaddr):
     assert a1.get_attr('received') == message
 
 
+def test_pubsub_without_serialization(nsaddr):
+    """
+    Simple publisher-subscriber pattern test.
+    """
+    a0 = run_agent('a0')
+    a1 = run_agent('a1')
+    a1.set_attr(received=None)
+    addr = a0.bind('PUB', alias='pub', serializer='raw')
+    a1.connect(addr, handler=set_received)
+    message = b'Hello world'
+    while not a1.get_attr('received'):
+        a0.send('pub', message)
+        time.sleep(0.1)
+    assert a1.get_attr('received') == message
+
+
 def test_agent_inheritance(nsaddr):
     """
     Test agent inheritance; agents can be based on a custom class.
```

Comment:
These tests are not well defined. Notice that, if you change `raw` by `pickle`, they all pass. So the functionality could be not implemented without failing (not saying that this is the case).

In order to test the `raw` sockets we could create a single agent and connect with it via a `zmq.Context().socket()` directly to test communications, instead of passing messages between two agents. That is more realistic as the typical use-case for `raw` sockets is when connecting from an external application using "raw" ZMQ.